onnxruntime/core/providers/tensorrt/tensorrt_provider_factory.cc

```diff
@@ -70,10 +71,61 @@ struct Tensorrt_Provider : Provider {
     return std::make_shared<TensorrtProviderFactory>(info);
   }
 
+  void UpdateInfo(void* provider_options, const ProviderOptions& options) override {
+    auto internal_options = onnxruntime::TensorrtExecutionProviderInfo::FromProviderOptions(options);
+    auto& trt_options = *reinterpret_cast<OrtTensorRTProviderOptions*>(provider_options);
+    trt_options.device_id = internal_options.device_id;
+    trt_options.trt_max_partition_iterations = internal_options.max_partition_iterations;
+    trt_options.trt_min_subgraph_size = internal_options.min_subgraph_size;
+    trt_options.trt_max_workspace_size = internal_options.max_workspace_size;
+    trt_options.trt_fp16_enable = internal_options.fp16_enable;
+    trt_options.trt_int8_enable = internal_options.int8_enable;
+
+    if (internal_options.int8_calibration_table_name.size() == 0) {
+      trt_options.trt_int8_calibration_table_name = nullptr;
+    } else {
+      trt_options.trt_int8_calibration_table_name = new char[internal_options.int8_calibration_table_name.size() + 1];
+      strcpy((char*)trt_options.trt_int8_calibration_table_name, internal_options.int8_calibration_table_name.c_str());
```

Comment:
there was a recent PR to update strlen to strnlen, I wonder if we should use strncpy as well?